modules/suitecrm.nix

```diff
@@ -0,0 +1,176 @@
+{ config, pkgs, lib, ... }:
+with lib;
+let
+  cfg = config.services.suitecrm;
+
+  phpFpmAddress =
+    if cfg.phpFpmAddress == null then
+      config.services.phpfpm.pools.suitecrm.socket
+    else
+      cfg.phpFpmAddress;
+in
+{
+  options.services.suitecrm = {
+    enable = mkEnableOption
+      "Enable SuiteCRM service, WARNING: will hijack the nginx service's suitecrm location.";
+
+    suitecrmPackage = mkOption {
+      type = with types; package;
+      description = ''
+        Which derivation contains the source of SuiteCRM, <litera>composor install</literal>,
+        must have already been run on the source
```

Comment:
I think a better way would just be to describe it as a package not as a source that's been transformed, prior art would be https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/element/element-web.nix which is a chat client in nixpkgs that contains no binaries but is something you can serve with nginx.

Reply:
I gave the description another shot